### PR TITLE
add pagebench test cases for periodic pagebench on dedicated hardware

### PR DESCRIPTION
**Pull request by Bodobolero**
we want to run some specific pagebench test cases on dedicated hardware to get reproducible results

run1: 1 client per tenant => characterize throughput with n tenants.
-  500 tenants
- scale 13 (200 MB database)
- 1 hour duration
- ca 380 GB layer snapshot files

run2.singleclient: 1 client per tenant => characterize latencies
run2.manyclient: N clients per tenant => characterize throughput scalability within one tenant.
- 1 tenant with 1 client for latencies
- 1 tenant with 64 clients because typically for a high number of connections we recommend the connection pooler
which by default uses 64 connections (for scalability)
- scale 136 (2048 MB database)
- 20 minutes each

### CODE REVIEW

File: test_runner/performance/pageserver/util.py
Comment (problame) [inline test_runner/performance/pageserver/util.py:25]: Is this still needed? 1 second per tenant seems plenty.

**Review by problame — CHANGES_REQUESTED** [review]
Can we remove `test_pageserver_max_throughput_getpage_at_latest_lsn` ?

And can we reduce the amount of copy-pasting in general?

Feels like
* `test_pageserver_max_throughput_getpage_at_latest_lsn`
* `test_pageserver_characterize_throughput_with_n_tenants`
* `test_pageserver_characterize_latencies_with_1_client_and_throughput_with_many_clients_one_tenant`

all have the exact same function body except `n_clients` 

=> these function bodies should be deduplicated into one helper function. The three functions above just call that helper function.

File: .github/workflows/periodic_pagebench.yml
Comment (Bodobolero) [reply .github/workflows/periodic_pagebench.yml:67]: modified the code

File: test_runner/performance/pageserver/util.py
Comment (Bodobolero) [reply test_runner/performance/pageserver/util.py:25]: yes it is needed for the 1 tenant with 1 client case, which failed if just waiting 1 second

File: .github/workflows/periodic_pagebench.yml
Comment (Bodobolero) [reply .github/workflows/periodic_pagebench.yml:69]: fixed

Comment (Bodobolero) [reply .github/workflows/periodic_pagebench.yml:69]: fixed

**Review by problame — APPROVED** [review]
Future readers will wonder what `max_throughput_latest_lsn` is, as it doesn't show up anywhere else. It would be nice to get rid of all occurences of that string, but, don't want to hold you up with this kind of stuff any longer than necessary. Approving and trusting you to apply judgement.

Regarding the github workflow: I don't see it doing `check-permissions`. See the `build_and_test.yml` that uses it. I think @bayandin should review those parts / security team should take a look.

**Comment by Bodobolero** [comment]
> I don't see it doing `check-permissions`

Not sure if we need this if it can only be run from manual dispatch and periodic runs, but not from triggers like fork commits.

File: test_runner/performance/pageserver/pagebench/test_pageserver_max_throughput_getpage_at_latest_lsn.py
Comment (Bodobolero) [reply test_runner/performance/pageserver/pagebench/test_pageserver_max_throughput_getpage_at_latest_lsn.py:201]: ok

File: .github/workflows/periodic_pagebench.yml
Comment (bayandin) [inline .github/workflows/periodic_pagebench.yml:28]: We can use `build-tools` image, then we won't need to do `Setup environment`, `Install AWS CLI`, and `Check AWS CLI version` steps:

```suggestion
        image: neondatabase/build-tools:pinned
```

File: .github/workflows/periodic_pagebench.yml
Comment (bayandin) [inline .github/workflows/periodic_pagebench.yml:29]: We need to use credentials to reduce the risk of being throttled:

```suggestion
        options: --init
        credentials:
          username: ${{ secrets.NEON_DOCKERHUB_USERNAME }}
          password: ${{ secrets.NEON_DOCKERHUB_PASSWORD }}
```

File: .github/workflows/periodic_pagebench.yml
Comment (bayandin) [inline .github/workflows/periodic_pagebench.yml:149]:
```suggestion
        aws ec2 wait instance-stopped --instance-ids $AWS_INSTANCE_ID

```

File: .github/workflows/periodic_pagebench.yml
Comment (bayandin) [inline .github/workflows/periodic_pagebench.yml:23]: To disallow running several runs of this workflow:

```suggestion

concurrency:
  group: ${{ github.workflow }}
  cancel-in-progress: false

```

**Review by bayandin — COMMENTED** [review]
I left a couple of suggestions regarding the pipeline.

Also, the indentation in the yaml is off a bit, in some places it uses 4 spaces, but we use 2 spaces (it would be nice to add a yaml linter in the future)

File: .github/workflows/periodic_pagebench.yml
Comment (Bodobolero) [reply .github/workflows/periodic_pagebench.yml:23]: This is already taken care off in the ec2 runner itself (2nd to nth run will just error out, but it is still a good idea to check it here, too)

File: .github/workflows/periodic_pagebench.yml
Comment (Bodobolero) [reply .github/workflows/periodic_pagebench.yml:29]: ok

File: .github/workflows/periodic_pagebench.yml
Comment (Bodobolero) [reply .github/workflows/periodic_pagebench.yml:28]: ok

**Review by bayandin — APPROVED** [review]
> Regarding the github workflow: I don't see it doing `check-permissions`. See the `build_and_test.yml` that uses it. I think @bayandin should review those parts / security team should take a look.

 `check-permissions` is not really a security measurement but mostly for our convenience — to avoid triggering many jobs that will fail due to a lack of access to permissions (if PR is opened by an external contributor).

Since this job can be triggered manually or by schedule, we don't need it.

**Pull request merged**
